controller/array_action/array_mediator_ds8k.py

```diff
@@ -217,20 +224,20 @@ def create_volume(self, name, size_in_bytes, capabilities, pool_id):
                 "Start to create volume with parameters: {}".format(cli_kwargs)
             )
 
-            try:
-                # get the volume before creating again, to make sure it is not existing,
-                # because volume name is not unique in ds8k.
-                volume = self.get_volume(
-                    name,
-                    pool_id=pool_id,
-                )
-                logger.info("Found volume {}".format(name))
-                return volume
-            except array_errors.VolumeNotFoundError:
-                volume = self.client.create_volume(**cli_kwargs)
+            # get the volume before creating again, to make sure it is not existing,
+            # because volume name is not unique in ds8k.
+            api_volume = self._get_api_volume_by_name(
+                name,
+                pool_id=pool_id
+            )
+            logger.info("Found volume {}".format(name))
+            if api_volume is not None:
+                return api_volume
```

Comment:
shouldn't it be `return self._generate_volume_response(api_volume)`?
if so, please covert this with a unit test